src/Extension/FluentExtension.php

```diff
@@ -679,6 +685,86 @@ public function onAfterDuplicate($original, $doWrite, $relations): void
             }
         }
+
+        // We don't have any localised relations to cover
+        if (count($copyHasOneRelations) === 0) {
+            return;
+        }
+
+        $currentLocale = FluentState::singleton()->getLocale();
+
+        // We don't have a current locale which indicates that the Fluent setup is incomplete - bail out
+        if (!$currentLocale) {
+            return;
+        }
+
+        $locales = [];
+
+        /** @var RecordLocale $localeInformation */
+        foreach ($owner->Locales() as $localeInformation) {
+            $sourceLocale = $localeInformation->getSourceLocale();
+            $modelLocale = $localeInformation->getLocaleObject();
+
+            if (!$sourceLocale) {
+                // We don't have any source locale, so we can bail out
+                continue;
+            }
+
+            if ($modelLocale->Locale !== $sourceLocale->Locale) {
+                // Source of this locale is different from current locale, so we can skip it
+                // as this locale content is being inherited
+                continue;
+            }
+
+            if ($modelLocale->Locale === $currentLocale) {
+                // Current locale can be skipped as it was already handled correctly
+                continue;
+            }
+
+            // Add locale which uses current locale as a source to our list
+            $locales[] = $modelLocale->Locale;
+        }
+
+        // No locales need to be actioned
+        if (count($locales) === 0) {
+            return;
+        }
+
+        foreach ($locales as $locale) {
+            FluentState::singleton()->withState(
+                static function (FluentState $state) use ($owner, $locale, $copyRelations): void {
+                    $state->setLocale($locale);
+
+                    $localisedOwner = DataObject::get($owner->ClassName)->byID($owner->ID);
+
+                    // Couldn't find localised data to work with
+                    if (!$localisedOwner->exists()) {
+                        return;
+                    }
+
+                    // Duplicate all localised relations
+                    foreach ($copyRelations as $relation) {
+                        $original = $localisedOwner->{$relation}();
+
+                        if (!$original instanceof DataObject) {
+                            continue;
+                        }
+
+                        if (!$original->exists()) {
+                            continue;
+                        }
+
+                        $duplicate = $original->duplicate();
+                        $localisedOwner->setComponent($relation, $duplicate);
+                    }
+
+                    // Update localised data (without version as this is considered a part of the duplication action)
+                    $localisedOwner->hasExtension(Versioned::class)
+                        ? $localisedOwner->writeWithoutVersion()
+                        : $localisedOwner->write();
```

Comment:
This might result in `makeLocalisedCopy()` being called in some cases - do we need to wrap this in `withLocalisedCopyState()` with `setLocalisedCopyActive(false)`?